File: September Leetcode Challenge/Day5/429. N-ary Tree Level Order Traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
     
        if(root == NULL)
        {
            return {};
        }
        queue<Node*>q;
        q.push(root);
        vector<vector<int>>ans;
        
        while(!q.empty())
        {
          int n=q.size();
          vector<int>v;
            
          for(int i=0; i<n; i++)
          {
              Node*temp = q.front();
              q.pop();
              for(auto child:temp->children)
              {
                  q.push(child);
              }
              v.push_back(temp->val);
          }
            ans.push_back(v);
            
        }
        return ans;
        
    }
};

File: September Leetcode Challenge/Day5/429. N-ary Tree Level Order Traversal.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>>ans = new ArrayList<List<Integer>>();
        if(root==null)
        {
            return ans;
        }
        Queue<Node>q = new LinkedList<>();
        q.offer(root);
        while(!q.isEmpty())
        {
            int n = q.size();
            ArrayList<Integer> v = new ArrayList<>();
            for(int i=0 ;i<n;i++)
            {
                for(Node child : q.peek().children)
                {
                    q.offer(child);
                }
                v.add(q.peek().val);
                q.poll();
                
            }
            ans.add(v);
         }
        return ans;
    }
}
